### PR TITLE
[Metrics] Add metrics to show the distribution of tablets version count

### PR DESCRIPTION
**Pull request by weizuo93**
## Proposed changes

 Add metrics (P50, P75, P90, P95, P99, etc.) to show the distribution of tablets version count.

```
# TYPE doris_be_tablet_version_num_distribution histogram
doris_be_tablet_version_num_distribution{quantile="0.50"} 9.21429
doris_be_tablet_version_num_distribution{quantile="0.75"} 11.7949
doris_be_tablet_version_num_distribution{quantile="0.90"} 13
doris_be_tablet_version_num_distribution{quantile="0.95"} 13
doris_be_tablet_version_num_distribution{quantile="0.99"} 13
doris_be_tablet_version_num_distribution_sum 950
doris_be_tablet_version_num_distribution_count 100
```

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)

### CODE REVIEW

File: be/src/util/doris_metrics.h [hunk 1]
```diff
@@ -137,6 +137,20 @@ class DorisMetrics {
...
+    UIntGauge* tablet_version_num_distribution_sum;
+    DoubleGauge* tablet_version_num_distribution_median;
+    DoubleGauge* tablet_version_num_distribution_average;
+    DoubleGauge* tablet_version_num_distribution_standard_deviation;
+    DoubleGauge* tablet_version_num_distribution_p50;
+    DoubleGauge* tablet_version_num_distribution_p75;
+    DoubleGauge* tablet_version_num_distribution_p90;
+    DoubleGauge* tablet_version_num_distribution_p95;
+    DoubleGauge* tablet_version_num_distribution_p99;
+
```
Comment (acelyc111) [inline be/src/util/doris_metrics.h:152]: Use HistogramMetric directly, its output is standard now.

Comment (weizuo93) [reply be/src/util/doris_metrics.h:152]: > Use HistogramMetric directly, its output is standard now.

OK,  I see it too. PR #5671 has been merged.

**Review by morningman — DISMISSED** [review]
LGTM

File: be/src/olap/tablet_manager.cpp [hunk 2]
```diff
@@ -976,13 +978,17 @@ OLAPStatus TabletManager::build_all_report_tablets_info(std::map<TTabletId, TTab
                     expire_txn_map.erase(find);
                 }
                 t_tablet.tablet_infos.push_back(tablet_info);
+                if (tablet_ptr->tablet_id() == tablet_id) {
+                    DorisMetrics::instance()->tablet_version_num_distribution->add(tablet_ptr->version_count());
```
Comment (acelyc111) [inline be/src/olap/tablet_manager.cpp:982]: It's not atomic and may last a long time before all values added, that means we may get a middle state which is not very correct.

Comment (weizuo93) [reply be/src/olap/tablet_manager.cpp]: > It's not atomic and may last a long time before all values added, that means we may get a middle state which is not very correct.

Thank you, you are right.

**Review by acelyc111 — DISMISSED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**